File: download/src/main/java/org/sqlunet/download/Chooser.java
/*
 * Copyright (c) 2022. Bernard Bou
 */

package org.sqlunet.download;

import android.util.Pair;
import android.widget.RadioButton;
import android.widget.RadioGroup;

import java.io.File;

import androidx.annotation.NonNull;
import androidx.fragment.app.FragmentActivity;

/**
 * Md5 async task
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
public class Chooser
{
	// static private final String TAG = "Chooser";

	/**
	 * Collect files
	 *
	 * @param directories directory names-values array
	 */
	@SafeVarargs
	static public RadioGroup collectFiles(@NonNull final FragmentActivity activity, final Pair<CharSequence[], CharSequence[]>... directories)
	{
		final RadioGroup group = new RadioGroup(activity);

		// collect targets
		for (Pair<CharSequence[], CharSequence[]> namesValues : directories)
		{
			final CharSequence[] names = namesValues.first;
			final CharSequence[] values = namesValues.second;
			for (int i = 0; i < names.length && i < values.length; i++)
			{
				//final CharSequence name = names[i];
				final CharSequence value = values[i];
				final String dirValue = value.toString();
				final File dir = new File(dirValue);
				if (!dir.exists())
				{
					continue;
				}
				//final File[] files = dir.listFiles((dir1, name) -> name.matches(".*\\.zip") || name.matches(".*\\.db"));
				//final File[] files = dir.listFiles((dir1, name) -> !new File(dir1, name).isDirectory());
				final File[] files = dir.listFiles();
				if (files == null)
				{
					continue;
				}
				for (File file : files)
				{
					if (file.isDirectory())
					{
						continue;
					}

					if (file.exists())
					{
						//final SpannableStringBuilder sb = new SpannableStringBuilder();
						//Report.appendHeader(sb, name.toString().split("\n")[0]).append('\n').append(file.getAbsolutePath());
						final RadioButton radioButton = new RadioButton(activity);
						radioButton.setText(file.getAbsolutePath());
						radioButton.setTag(file.getAbsolutePath());
						group.addView(radioButton);
					}
				}
			}
		}
		return group.getChildCount() == 0 ? null : group;
	}
}

File: download_common/src/main/java/com/bbou/download/choose/Chooser.kt
/*
 * Copyright (c) 2023. Bernard Bou <1313ou@gmail.com>.
 */
package com.bbou.download.choose

import android.annotation.SuppressLint
import android.content.Context
import android.widget.RadioButton
import android.widget.RadioGroup
import java.io.File

/**
 * Chooser
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
object Chooser {

    /**
     * Collect files into radio group
     *
     * @param context context
     * @param directories directories
     * @return radio group
     */
    @SafeVarargs
    fun toRadioGroup(context: Context, directories: Array<String>): RadioGroup? {
        val names = directories.map { File(it).name }.toTypedArray()
        return toRadioGroup(context, names, directories)
    }

    /**
     * Collect files into radio group
     *
     * @param context context
     * @param names names
     * @param values values
     * @return radio group
     */
    @SuppressLint("SetTextI18n")

    @SafeVarargs
    fun toRadioGroup(context: Context, names: Array<out CharSequence>, values: Array<String>): RadioGroup? {
        val group = RadioGroup(context)
        var i = 0
        while (i < values.size && i < names.size) {

            val name = names[i]
            val value = values[i]
            val dir = File(value)
            if (dir.exists()) {

                // files in dir
                val files = dir.listFiles()
                if (files != null) {
                    for (file in files) {
                        if (!file.isDirectory && file.exists()) {
                            val radioButton = RadioButton(context)
                            radioButton.text = "${name}/${file.name}"
                            radioButton.tag = file.absolutePath
                            group.addView(radioButton)
                        }
                    }
                }
            }
            i++
        }
        return if (group.childCount == 0) null else group
    }
}
